src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,90 @@
+package codeu.chat.util;
+
+import java.lang.StringBuilder;
+import java.io.IOException;
+
+// This utility class allows us to read input that may contain
+// multiple tokens to allow for multi-token commands in our
+// application later on.
+public final class Tokenizer {
+
+  // The following instance variables help keep track of the current token
+  // we are building (token), the original input (source), and the current
+  // index we are at in our source (at).
+  private StringBuilder token;
+  private String source;
+  private int at;
```

Comment:
Totally understandable if you were following the guide, but I'm glad you spotted the opacity.